### ISSUE TITLE
Exceptions captured through `Extensions.Logging` marked as `handled = true`

### ISSUE THREAD

**Issue by bitsandfoxes** [getsentry/sentry-dotnet#3394]
An easy repro is `Sentry.Samples.AspNetCore.Blazor.Wasm`: https://github.com/getsentry/sentry-dotnet/blob/721b411a8ddc87b8e622ea4cf86949748511c8d9/samples/Sentry.Samples.AspNetCore.Blazor.Wasm/Pages/Thrower.razor#L24
The sample `throws null` but the error gets captured through `Sentry.Extensions.Logging` which calls `_hub.CaptureException` which ends up in here getting marked as `handled` https://github.com/getsentry/sentry-dotnet/blob/721b411a8ddc87b8e622ea4cf86949748511c8d9/src/Sentry/Internal/MainExceptionProcessor.cs#L177-L182

**Comment by jamescrosswell** [comment]
This seems like a bit of an edge case - it's pretty unusual to `throw null`. Maybe we leave the logic for the SDK as it is and update the sample to throw an exception instead?

**Comment by bitsandfoxes** [comment]
It's not about the `throw null` but about our logging integration capturing an exception it receives from the logs and the processor assuming "if exception has stacktrace then handled`.

**Comment by bitsandfoxes** [comment]
Since we're fairly sure that this is most likely relevant in a WASM context, figuring out how to properly set up global exception handling might make this problem go away on its own. [MS Docs](https://learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/handle-errors?view=aspnetcore-8.0#error-boundaries)

**Comment by jamescrosswell** [comment]
Tricky. When an unhandled exception occurs in WASM... the [framework logs these](https://github.com/dotnet/aspnetcore/blob/7df092d1d13692097507fcfe2c1eecdb923daa08/src/Components/WebAssembly/WebAssembly/src/Rendering/WebAssemblyRenderer.cs#L167-L180). The logging itself is taken care of by various source generated helper functions like this one [for unhandled exceptions that occur during rendering](https://github.com/dotnet/aspnetcore/blob/7df092d1d13692097507fcfe2c1eecdb923daa08/src/Components/WebAssembly/WebAssembly/src/Rendering/WebAssemblyRenderer.cs#L189-L193).

Potentially, we could take advantage of the fact that the LogMessageAttribute (and the corresponding [source generator](https://learn.microsoft.com/en-us/dotnet/core/extensions/logger-message-generator) that is using this marker attribute) is setting an EventId... in the case of unhandled rendering errors it's 100:
```csharp
    private static partial class Log
    {
        [LoggerMessage(100, LogLevel.Critical, "Unhandled exception rendering component: {Message}", EventName = "ExceptionRenderingComponent")]
        public static partial void UnhandledExceptionRenderingComponent(ILogger logger, string message, Exception exception);
    }
```

There are various [places where an error may occur](https://learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/handle-errors?view=aspnetcore-8.0#places-where-errors-may-occur) and each of these may result in a different EventId, so we'd have to do a bit of rummaging around in the source code to compile a list of all the UnhandledException EventIds used by Blazor. Theoretically, this could work though.

**Comment by jamescrosswell** [comment]
Potential (partial) fix here: https://github.com/getsentry/sentry-dotnet/pull/3386/commits/c6ccc52e840f27943999c509407ac6ae08343957. It will identify the exception from our Sample project but I'm sure there are all kinds of other scenarios where the `EventId.Name` might be something else. Probably need to do a bit more testing, trying to generate errors from the various other places where an error may occur (linked above).

**Comment by bruno-garcia** [comment]
we could pass some scope to the logger that the sdk understands as setting the event as unhandled. like logError<SentryUnhandled>(...)
or set something in the scope instead?

**Comment by bruno-garcia** [comment]
> Potential (partial) fix here: c6ccc52

those strings are guaranteed to be unhanlded/framework captured exceptions ?

feels fragile that way, but if we're not the ones calling logException i don't see another way than using the logger id

**Comment by jamescrosswell** [comment]
Fixed in https://github.com/getsentry/sentry-dotnet/pull/3386
